### PR TITLE
closes #36 Update of the test data with the values for children

### LINKED ISSUE #36: Update our test data to bring in line with the CDISC test pilot dummy subjects
The test data will need updating to bring in line with the CDISC test pilot dummy subjects from {pharmaversesdtm} in the following ways:

- [ ] `dm_peds` should be a copy of the current {pharmaversesdtm} `dm` data but dates of birth changed to make these a range of children's ages (whereas pharmaversesdtm test data comes from an adults study) - birth dates should all be complete (no partial) as not for us to recommend a rule to do in this case.
- [ ] similarly `vs_peds` should use {pharmaversesdtm} `vs` as a starting point ensuring we have only height, weight, bmi and head circumference parameters and these are made more realistic for childrens levels and the assessment dates need to be made more realistic to align with each child's birth dates. e.g. don't keep VS dates < DM birth dates or it won't make sense when we come to derive the current age (as it would come out negative).

### PR DESCRIPTION
**Pull request by remikk**
Update of birthdate to represent child subject; complete partial date; update height, weight and bmi to values typical for children.

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiralpeds (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiralpeds site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the Reference page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by rossfarrugia** [comment]
@remikk I think you missed the intention of this issue so i've tried to re-write the issue more clearly here: https://github.com/pharmaverse/admiralpeds/issues/36. I was hoping to see code such as the following for `dm_peds`:

```
library(dplyr)
library(admiral)
library(pharmaversesdtm)

# Read input test data from pharmaversesdtm ----
data("dm")

# Convert blank to NA ----
dm <- convert_blanks_to_na(dm)

# Subset to first 5 patients only (which is enough for our examples) ----
dm_subset <- dm %>%
  filter(USUBJID %in% c("01-701-1015", "01-701-1023", "01-701-1028",
                        "01-701-1033", "01-701-1034"))

# Add birth dates/age realistic for pediatrics in line with treatment dates ----
dm_peds <- dm_subset %>%
  mutate(BRTHDTC = case_when(
    USUBJID == "01-701-1015" ~ "2013-01-02",
    USUBJID == "01-701-1023" ~ "2010-08-05",
    USUBJID == "01-701-1028" ~ "2010-07-19",
    USUBJID == "01-701-1033" ~ "2014-01-01",
    USUBJID == "01-701-1034" ~ "2014-06-01"
  )) %>%
  mutate(AGE = case_when(
    USUBJID == "01-701-1015" ~ 1,
    USUBJID == "01-701-1023" ~ 2,
    USUBJID == "01-701-1028" ~ 3,
    USUBJID == "01-701-1033" ~ 0,
    USUBJID == "01-701-1034" ~ 0
  ))

# Variable labels ----
attr(dm_peds$BRTHDTC, "label") <- "Date/Time of Birth"

# Label dataset ----
attr(dm_peds, "label") <- "Demographics"

# Save dataset ----
usethis::use_data(dm_peds, overwrite = TRUE)
```

The reason I used dates from 2010 - 2014 is aligning with the patients dates in {pharmaversesdtm} dm data, as if we make these dates work then you shouldn't have to subsequently update the `vs` dates. You'll just need to update the VSORRES/VSSTRESC/VSSTRESN to make the readings more realistic to heights/weights/bmi of children.

Please try this out and if you agree commit these updates and then try similar for `vs_peds`.

File: data-raw/vs_peds.R [hunk 1]
```diff
@@ -2,160 +2,218 @@
...
+      VISIT == "WEEK 20" ~ 41.89,
+      VISIT == "WEEK 24" ~ 42.30,
+      VISIT == "WEEK 26" ~ 42.72
+    ),
+    TRUE ~ VSSTRESN
+  )) %>%
+  arrange(USUBJID, VISITNUM, VSTESTCD, VSDY) %>%
+  # vs_subset_calc <- vs_subset_full
+
+  group_by(USUBJID, VISITNUM) %>%
```
Comment (rossfarrugia) [inline data-raw/vs_peds.R:204]: i would make this a separate step with its own dataframe and comment associated to explain that BMI vaues are being derived from weight/height, as breaking up the code into clear blocks makes it more modular and easier to maintain in future if ever needing updates. also you could try using https://pharmaverse.github.io/admiral/reference/derive_param_bmi.html here if you like from admiral to derive the BMI.

File: data-raw/vs_peds.R [hunk 2]
```diff
@@ -183,11 +241,7 @@ vs_peds <- tibble::tribble(
     VSEVAL = NA_character_,
     EPOCH = "Epoch"
   ) %>%
-  select(
-    STUDYID, DOMAIN, USUBJID, VSSEQ, VSTESTCD, VSTEST,
-    VSORRES, VSORRESU, VSSTRESC, VSSTRESN, VSSTRESU, VSEVAL,
-    EPOCH, VISIT, VISITNUM, VSDTC, VSDY
-  )
+  arrange(USUBJID, VISITNUM, VSTESTCD, VSDY)
```
Comment (rossfarrugia) [inline data-raw/vs_peds.R:244]: no need to re sort at this stage - remove

File: data-raw/vs_peds.R [hunk 3]
```diff
@@ -183,11 +241,7 @@ vs_peds <- tibble::tribble(
...
 
 # get common column names with VS
```
Comment (rossfarrugia) [inline data-raw/vs_peds.R:246]: this time this block does seem to be needed as `dplyr::bind_rows()` clears the labels but when i run this block it doesn't seem to keep the labels. maybe use `metatools::add_labels()` to set the variable labels instead like in for example https://github.com/pharmaverse/pharmaversesdtm/blob/main/data-raw/ce_vaccine.R

File: data-raw/vs_peds.R [hunk 4]
```diff
@@ -2,160 +2,218 @@
...
+      VISIT == "SCREENING 1" ~ 87.65,
+      VISIT == "BASELINE" ~ 88.25,
+      VISIT == "WEEK 2" ~ 88.75,
+      VISIT == "WEEK 4" ~ 89.23
+    ),
+    USUBJID == "01-701-1023" & VSTESTCD == "HDCIRC" ~ case_when(
+      VISIT == "SCREENING 1" ~ 87.65,
+      VISIT == "BASELINE" ~ 88.25,
+      VISIT == "WEEK 2" ~ 88.75,
+      VISIT == "WEEK 4" ~ 89.23
```
Comment (rossfarrugia) [inline data-raw/vs_peds.R:101]: Some of the head circumferences seem extremely high - i would make them maximum 50 as per the medians from https://github.com/pharmaverse/admiralpeds/blob/main/data-raw/who_hc_for_age_boys.R

**Review by rossfarrugia — CHANGES_REQUESTED** [review]
@remikk thanks for the efforts! looks really good. here's my mostly minor comments and then happy to get this one merged.

File: data-raw/vs_peds.R [hunk 5]
```diff
@@ -2,160 +2,218 @@
...
-  "PEDS-1009", "6 MONTHS", 3, "WEIGHT", "2024-07-02", 112.651415,
-  "PEDS-1009", "6 MONTHS", 3, "HEIGHT", "2024-07-02", 167.320201,
-  "PEDS-1009", "6 MONTHS", 3, "BMI", "2024-02-07", 40.23833,
-  "PEDS-1009", "6 MONTHS", 3, "HDCIRC", "2024-07-02", NA,
-  "PEDS-1009", "12 MONTHS", 4, "WEIGHT", "2024-12-02", 156,
-  "PEDS-1009", "12 MONTHS", 4, "HEIGHT", "2024-12-02", 168,
-  "PEDS-1009", "12 MONTHS", 4, "BMI", "2024-02-12", 55.2721,
-  "PEDS-1009", "12 MONTHS", 4, "HDCIRC", "2024-12-02", NA
+# Convert blank to NA ----
+# vs <- convert_blanks_to_na(vs)
```
[the remark this answers is not in the document]
Comment (remikk) [reply data-raw/vs_peds.R:13]: The conversion has been uncommented.

File: data-raw/dm_peds.R [hunk 6]
```diff
@@ -2,66 +2,42 @@
 # Description: Create DM test SDTM dataset for pediatric studies
```
[the remark this answers is not in the document]
Comment (remikk) [reply data-raw/dm_peds.R:2]: The spelling has been corrected.

File: data-raw/vs_peds.R [hunk 7]
```diff
@@ -2,160 +2,218 @@
...
+
+# Subset to 5 patients present in dm_peds ----
```
[the remark this answers is not in the document]
Comment (remikk) [reply data-raw/vs_peds.R:15]: The comment has been updated accordingly.

File: data-raw/vs_peds.R [hunk 8]
```diff
@@ -2,160 +2,218 @@
...
+      VISIT == "WEEK 6" ~ 40.23,
+      VISIT == "WEEK 8" ~ 40.65,
+      VISIT == "WEEK 12" ~ 41.07,
+      VISIT == "WEEK 16" ~ 41.48,
+      VISIT == "WEEK 20" ~ 41.89,
+      VISIT == "WEEK 24" ~ 42.30,
+      VISIT == "WEEK 26" ~ 42.72
+    ),
+    TRUE ~ VSSTRESN
+  )) %>%
```
[the remark this answers is not in the document]
Comment (remikk) [reply data-raw/vs_peds.R:200]: It has been updated as suggested.

File: data-raw/vs_peds.R (hunk 1 above)
Comment (remikk) [reply data-raw/vs_peds.R:204]: Thank you for the comment. Fully agree. The code has been broken down into functional sections with appropriate comments for BMI calculation. I have, however, stayed with the previously coded derivation. Unless there was a simpler way to do that, derive_param_bmi required me to rename existing VSTESTCD and VSSTRESN, or create a set of working variables like PARAMCD and AVAL, and drop prepared placeholder records for the BMI parameter. 
Should you find it more appropriate to use the function rather than keeping existing code I will do that. Please advise. Thank you.

File: data-raw/vs_peds.R (hunk 2 above)
Comment (remikk) [reply data-raw/vs_peds.R:244]: It has been removed.

File: data-raw/vs_peds.R (hunk 4 above)
Comment (remikk) [reply data-raw/vs_peds.R:101]: Thank you for the finding. That was my copy-paste mistake. I copied the section with height values, then updated the pram name and missed updating the values. The issue has been corrected.

File: data-raw/dm_peds.R [hunk 9]
```diff
@@ -70,8 +46,6 @@ lapply(common_cols, function(x) {
   attr(dm_peds[[common_cols[x]]], "label") <- attr(dm[[common_cols[x]]], "label")
 })
```
[the remark this answers is not in the document]
Comment (remikk) [reply data-raw/dm_peds.R:47]: Indeed. This section of the code has been removed with the preceding '# get common column names' part. It did not work anyway, as AGE was missing the label. I added a line for setting this label in the 'Variable labels' section.

File: data-raw/vs_peds.R [hunk 10]
```diff
@@ -2,160 +2,218 @@
...
+vs_subset <- vs %>%
+  filter(USUBJID %in% c(
+    "01-701-1015", "01-701-1023", "01-701-1028",
+    "01-701-1033", "01-701-1034"
+  ) &
+    VSTESTCD %in% c("WEIGHT"))
+
+# Filter 'WEIGHT' records for placeholder for HEIGHT and HDCIRC
+vs_subset_height <- vs_subset %>%
+  filter(VSTESTCD == "WEIGHT") %>%
```
[the remark this answers is not in the document]
Comment (remikk) [reply data-raw/vs_peds.R:25]: Indeed. The filter has been removed.

File: data-raw/vs_peds.R [hunk 11]
```diff
@@ -2,160 +2,218 @@
...
+
+# Subset to 5 patients present in dm_peds ----
+vs_subset <- vs %>%
+  filter(USUBJID %in% c(
+    "01-701-1015", "01-701-1023", "01-701-1028",
+    "01-701-1033", "01-701-1034"
+  ) &
+    VSTESTCD %in% c("WEIGHT"))
+
+# Filter 'WEIGHT' records for placeholder for HEIGHT and HDCIRC
```
[the remark this answers is not in the document]
Comment (remikk) [reply data-raw/vs_peds.R:23]: It has been updated as suggested.

File: data-raw/vs_peds.R [hunk 12]
```diff
@@ -2,160 +2,218 @@
...
+  mutate(VSTESTCD = "HDCIRC")
+
+# Bind new HEIGHT records to original dataset
+vs_subset_full <- bind_rows(
+  vs_subset,
+  vs_subset_height,
+  vs_subset_bmi,
+  vs_subset_hdcirc
 ) %>%
+  arrange(USUBJID, VISITNUM, VSTESTCD, VSDY)
```
[the remark this answers is not in the document]
Comment (remikk) [reply data-raw/vs_peds.R:41]: Right. It has been updated accordingly.

File: data-raw/vs_peds.R [hunk 13]
```diff
@@ -2,160 +2,218 @@
...
+
+# Updating subject 01-701-1015 with data for 1 year old girl
```
[the remark this answers is not in the document]
Comment (remikk) [reply data-raw/vs_peds.R:43]: It has been updated accordingly.

File: data-raw/vs_peds.R [hunk 14]
```diff
@@ -2,160 +2,218 @@
...
+      VISIT == "WEEK 12" ~ 41.07,
+      VISIT == "WEEK 16" ~ 41.48,
+      VISIT == "WEEK 20" ~ 41.89,
+      VISIT == "WEEK 24" ~ 42.30,
+      VISIT == "WEEK 26" ~ 42.72
+    ),
+    TRUE ~ VSSTRESN
+  )) %>%
+  arrange(USUBJID, VISITNUM, VSTESTCD, VSDY) %>%
+  # vs_subset_calc <- vs_subset_full
```
[the remark this answers is not in the document]
Comment (remikk) [reply data-raw/vs_peds.R:202]: It has been removed.

File: data-raw/vs_peds.R [hunk 15]
```diff
@@ -2,160 +2,218 @@
...
+      VISIT == "WEEK 8" ~ 40.65,
+      VISIT == "WEEK 12" ~ 41.07,
+      VISIT == "WEEK 16" ~ 41.48,
+      VISIT == "WEEK 20" ~ 41.89,
+      VISIT == "WEEK 24" ~ 42.30,
+      VISIT == "WEEK 26" ~ 42.72
+    ),
+    TRUE ~ VSSTRESN
+  )) %>%
+  arrange(USUBJID, VISITNUM, VSTESTCD, VSDY) %>%
```
[the remark this answers is not in the document]
Comment (remikk) [reply data-raw/vs_peds.R:201]: It has been removed.

File: data-raw/vs_peds.R (hunk 3 above)
Comment (remikk) [reply data-raw/vs_peds.R:246]: The issue has been resolved by updating the current code for setting the labels for variables comm with vs and separately adding labels for two newly added variables, VSEVAL and EPOCH.

File: data-raw/vs_peds.R (hunk 1 above)
Comment (rossfarrugia) [reply data-raw/vs_peds.R:202]: thats fine Remi!

**Review by rossfarrugia — APPROVED** [review]
